argument.go

```diff
@@ -1,29 +1,23 @@
 package graphb
 
 import (
+	"bytes"
 	"fmt"
 )
 
 type argumentValue interface {
-	stringChan() <-chan string
+	stringChan(buffer *bytes.Buffer)
```

Comment:
This is fantastic. I wanted to do this when I first created it. But I was too happy about the clever use of channels & goroutine so that I left it there. But I do think an `io.Writer` alike version is better.

Some suggestions:
1. How about `stringWriter() *bytes.Buffer` which returns an writer instead of consuming one?
2. Have you think about `io.Writer`?

Reply:
The function name could be `stringWriter` so that people don't get confused.